api/src/test/java/com/epam/pipeline/manager/pipeline/RunScheduleManagerTest.java

```diff
@@ -103,179 +121,242 @@ public void setUp() {
+
         runConfiguration = createRunConfiguration();
+        runConfiguration.setId(2L);
 
         PipelineUserVO userVO = new PipelineUserVO();
         userVO.setUserName(USER_OWNER);
-        userManager.createUser(userVO);
 
         testRunScheduleVO = getRunScheduleVO(RunScheduledAction.PAUSE, CRON_EXPRESSION1);
         testRunScheduleVO2 = getRunScheduleVO(RunScheduledAction.RESUME, CRON_EXPRESSION2);
         testRunScheduleVO3 = getRunScheduleVO(RunScheduledAction.RUN, CRON_EXPRESSION2);
 
+        MockitoAnnotations.initMocks(this);
     }
 
     @Test
-    @WithMockUser(username = USER_OWNER)
     public void testCreateRunSchedules() {
-        final List<RunSchedule> runSchedules = runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
-                    Arrays.asList(testRunScheduleVO, testRunScheduleVO2));
-        runSchedules.forEach(this::loadAndAssertSchedule);
+        when(mockRunScheduleDao.loadAllRunSchedules()).thenReturn(Collections.emptyList());
+        when(mockScheduleProviderManager.getProvider(ScheduleType.PIPELINE_RUN)).thenReturn(mockPipelineRunScheduleProvider);
+        when(mockScheduleProviderManager.getProvider(ScheduleType.RUN_CONFIGURATION)).thenReturn(mockRunConfigurationScheduleProvider);
+        when(mockAuthManager.getCurrentUser()).thenReturn(mockPipelineUser);
 
-        List<RunSchedule> configSchedules = runScheduleManager.createSchedules(runConfiguration.getId(),
-                ScheduleType.RUN_CONFIGURATION, Collections.singletonList(testRunScheduleVO3));
-        configSchedules.forEach(this::loadAndAssertSchedule);
+        runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
+                Arrays.asList(testRunScheduleVO, testRunScheduleVO2));
 
-        Mockito.verify(runScheduler, Mockito.times(3)).scheduleRunSchedule(any());
+        runScheduleManager.createSchedules(runConfiguration.getId(),
+                ScheduleType.RUN_CONFIGURATION, Collections.singletonList(testRunScheduleVO3));
 
+        verify(mockRunScheduler, times(3)).scheduleRunSchedule(any());
+        verify(mockRunScheduleDao, times(2)).createRunSchedules(anyList());
+        verify(mockScheduleProviderManager, times(3)).getProvider(ScheduleType.PIPELINE_RUN);
+        verify(mockScheduleProviderManager, times(2)).getProvider(ScheduleType.RUN_CONFIGURATION);
     }
 
+
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
     public void testCreateRunScheduleWithExistentCronExpression() {
+        when(mockScheduleProviderManager.getProvider(ScheduleType.PIPELINE_RUN)).thenReturn(mockPipelineRunScheduleProvider);
+        when(mockAuthManager.getCurrentUser()).thenReturn(mockPipelineUser);
+
         runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
                 Collections.singletonList(testRunScheduleVO));
         testRunScheduleVO.setAction(RunScheduledAction.RESUME);
```

Comment:
If you delete these lines the test still passes. You see, the problem here is that previously the test used real data base and some additional actions were required. As long as we mock everything now we can skip most of the preconfiguration and just test exactly what we want to verify.